src/OrchardCore/OrchardCore.Localization.Core/LocalizationManager.cs

```diff
@@ -1,66 +1,64 @@
+    public LocalizationManager(
+        IEnumerable<IPluralRuleProvider> pluralRuleProviders,
+        IEnumerable<ITranslationProvider> translationProviders,
+        IMemoryCache cache)
     {
-        private const string CacheKeyPrefix = "CultureDictionary-";
-
-        private static readonly PluralizationRuleDelegate _defaultPluralRule = n => (n != 1 ? 1 : 0);
-
-        private readonly IList<IPluralRuleProvider> _pluralRuleProviders;
-        private readonly IEnumerable<ITranslationProvider> _translationProviders;
-        private readonly IMemoryCache _cache;
+        _pluralRuleProviders = pluralRuleProviders.OrderBy(o => o.Order).ToArray();
+        _translationProviders = translationProviders;
+        _cache = cache;
+    }
 
-        /// <summary>
-        /// Creates a new instance of <see cref="LocalizationManager"/>.
-        /// </summary>
-        /// <param name="pluralRuleProviders">A list of <see cref="IPluralRuleProvider"/>s.</param>
-        /// <param name="translationProviders">The list of available <see cref="ITranslationProvider"/>.</param>
-        /// <param name="cache">The <see cref="IMemoryCache"/>.</param>
-        public LocalizationManager(
-            IEnumerable<IPluralRuleProvider> pluralRuleProviders,
-            IEnumerable<ITranslationProvider> translationProviders,
-            IMemoryCache cache)
+    /// <inheritdoc />
+    public async Task<CultureDictionary> GetDictionaryAsync(CultureInfo culture)
+    {
+        var cachedDictionary = await _cache.GetOrCreateAsync(CacheKeyPrefix + culture.Name, async (e) =>
         {
-            _pluralRuleProviders = pluralRuleProviders.OrderBy(o => o.Order).ToArray();
-            _translationProviders = translationProviders;
-            _cache = cache;
-        }
+            var rule = _defaultPluralRule;
 
-        /// <inheritdoc />
-        public CultureDictionary GetDictionary(CultureInfo culture)
-        {
-            var cachedDictionary = _cache.GetOrCreate(CacheKeyPrefix + culture.Name, k => new Lazy<CultureDictionary>(() =>
+            foreach (var provider in _pluralRuleProviders)
             {
-                var rule = _defaultPluralRule;
-
-                foreach (var provider in _pluralRuleProviders)
+                if (provider.TryGetRule(culture, out rule))
                 {
-                    if (provider.TryGetRule(culture, out rule))
-                    {
-                        break;
-                    }
+                    break;
                 }
+            }
 
-                var dictionary = new CultureDictionary(culture.Name, rule ?? _defaultPluralRule);
-                foreach (var translationProvider in _translationProviders)
-                {
-                    translationProvider.LoadTranslations(culture.Name, dictionary);
-                }
+            var dictionary = new CultureDictionary(culture.Name, rule ?? _defaultPluralRule);
```

Comment:
The `LoadTranslationsAsync` should be move to a private `AwaitedLoadtranslationAsync` method with the async/await keywords such that we don't need it in the `GetDictionaryAsync`.